docs/DeveloperGuide.adoc

```diff
@@ -1,93 +1,1101 @@
+. The original exam is added to the specific person.
+. For both the AddressBook and ExamBook, any copies of the original exam is removed and replaced with the new exam.
+
+This is demostrated by the following sequence diagram:
+[[fig-registerPersonDiagram]]
+.Sequence Diagram of registering person
+image::registerperson.png[width="800"]
+
+==== Design Considerations
+
+===== Aspect: How the exam data are being synchronised between the AddressBook and ExamBook
+
+* **Alternative 1 (current choice):** Each time there is a change in a field of an Exam, there has to be iteration through the whole AddressBook or ExamBook to update them. 
+** Pros: Less space is needed.
+** Cons: This can waste some time, if no changes need to be made for example.
+* **Alternative 2:** Store the persons registered for an exam for each exam.
+** Pros: This makes it faster to update any changes if needed.
+** Cons: More space is needed.
+
+===== Aspect: How the exam data are being updated
+
+* **Alternative 1 (current choice):** Each time there is a change in a field of an Exam, a new Exam is created with the new details and added into the ExamBook and the old Exam is deleted from the ExamBook.
+** Pros: Allows exams to be encapsulated using interface.
+** Cons: Temporary increase in space for object created during method .
+* **Alternative 2:** Have no interface.
+** Pros: Easy access to set values.
+** Cons: No encapsulation.
+
+===== Aspect: Data structure to support the exam commands
+
+* **Alternative 1 (current choice):** Exams are stored in a set under each Person object in the AddressBook but stored as a list in the ExamBook.
+** Pros: In the Exambook, it is required to access an exam through an index to delete and edit, hence list is better. In the AddressBook, exams are only to be used for viewing and accessed to check for value. 
+** Cons: Exams stored under each Person in the AddressBook cannot be sorted.
+* **Alternative 2:** Exams are stored as a list in both ExamBook and AddressBook.
+** Pros: Exams stored under each Person in the AddressBook can be sorted.
+** Cons: It will take longer to check if a Person has a same Exam with the Exam to be updated.
+
+===== Aspect: Storage of ExamBook
+
+* **Alternative 1 (current choice):** Exams are stored in a separate txt file from AddressBook
+** Pros: This allows for clarity and easy access to find and see exams in raw XML format. 
+** Cons: Extra storage file and hence methods and file paths are needed, this may be confusing and cause much repetition of code.
+* **Alternative 2:** Exams are stored in same data file as AddressBook.
+** Pros: Only 1 storage file is needed. Less repetition and variables are needed.
+** Cons: May be harder for user to see exams in raw XML format.
+// end::exam[]
+
+// tag::privilege[]
+=== Privilege Feature
+[[fig-PrivilegeDiagram]]
+.Structure of the Privilege Component
+image::Privilege.png[width="800"]
```

Comment:
Is this an architecture diagram/component/class diagram? It seems you have included notation for all. Also, does person belong to the Logic Component? 

You may want to consider creating an architecture diagram to explain the high-level design of your solution, then reverse-engineer a class diagram from code.  Note that you do not need to specify low level details in your class diagram to the extent that your diagram becomes big and clutter. 
 
Refer to the website on the design process again for more clarification: 

![screenshot 2018-10-19 18 05 22](https://user-images.githubusercontent.com/2998854/47211980-9070d100-d3c9-11e8-9bdd-670b9de7e910.png)